src/ast.rs

```diff
@@ -194,249 +152,152 @@ pub enum TypeScriptReference {
-    )
-    .into_iter()
-    .filter(|desc| desc.kind != DependencyKind::Require)
-    .collect()
-  }
-
-  /// Searches comments for any `@deno-types` compiler hints.
-  fn analyze_deno_types(
-    &self,
-    desc: &DependencyDescriptor,
-  ) -> Option<(String, Range)> {
-    let comment = desc.leading_comments.last()?;
-    let captures = DENO_TYPES_RE.captures(&comment.text)?;
-    if let Some(m) = captures.get(1) {
-      Some((
-        m.as_str().to_string(),
-        Range::from_comment_match(comment, &m, |pos| self.get_position(pos)),
-      ))
-    } else if let Some(m) = captures.get(2) {
-      Some((
-        m.as_str().to_string(),
-        Range::from_comment_match(comment, &m, |pos| self.get_position(pos)),
-      ))
-    } else {
-      unreachable!("Unexpected captures from deno types regex")
-    }
-  }
-
-  /// Searches comments for any triple slash references.
-  fn analyze_ts_references(&self) -> Vec<TypeScriptReference> {
-    let mut references = Vec::new();
-    for comment in self.get_leading_comments().iter() {
-      if TRIPLE_SLASH_REFERENCE_RE.is_match(&comment.text) {
-        if let Some(captures) = PATH_REFERENCE_RE.captures(&comment.text) {
-          let m = captures.get(1).unwrap();
-          references.push(TypeScriptReference::Path(
-            m.as_str().to_string(),
-            Range::from_comment_match(comment, &m, |pos| {
-              self.get_position(pos)
-            }),
-          ));
-        } else if let Some(captures) =
-          TYPES_REFERENCE_RE.captures(&comment.text)
-        {
-          let m = captures.get(1).unwrap();
-          references.push(TypeScriptReference::Types(
-            m.as_str().to_string(),
-            Range::from_comment_match(comment, &m, |pos| {
-              self.get_position(pos)
-            }),
-          ));
-        }
+/// Searches comments for any triple slash references.
+pub fn analyze_ts_references(
+  parsed_source: &ParsedSource,
+) -> Vec<TypeScriptReference> {
+  let mut references = Vec::new();
+  for comment in parsed_source.get_leading_comments().iter() {
+    if TRIPLE_SLASH_REFERENCE_RE.is_match(&comment.text) {
+      if let Some(captures) = PATH_REFERENCE_RE.captures(&comment.text) {
+        let m = captures.get(1).unwrap();
+        references.push(TypeScriptReference::Path(
+          m.as_str().to_string(),
+          Range::from_comment_match(comment, &m, parsed_source),
+        ));
+      } else if let Some(captures) = TYPES_REFERENCE_RE.captures(&comment.text)
+      {
+        let m = captures.get(1).unwrap();
+        references.push(TypeScriptReference::Types(
+          m.as_str().to_string(),
+          Range::from_comment_match(comment, &m, parsed_source),
+        ));
       }
     }
-    references
   }
+  references
 }
 
-/// Parses modules.
-pub trait AstParser {
-  /// Parses the provided information to a `ParsedAst` and returns a reference.
+/// Parses text to a `ParsedSource`.
+pub trait SourceParser {
```

Comment:
I thought about moving this trait down to `deno_ast`, but this has `deno_graph` specific stuff in it like `ModuleSpecifier` and it is opinionated about parsing a module rather than a program (possibly a script)